python/ray/experimental/channel.py

```diff
@@ -88,33 +98,64 @@ def __init__(
         if not isinstance(num_readers, int):
             raise ValueError("num_readers must be an integer")
 
+        self._buffer_size_bytes = buffer_size_bytes
+
+        self._reader_node_id = _reader_node_id
+        self._writer_channel = _writer_channel
+
         self._num_readers = num_readers
         self._worker = ray._private.worker.global_worker
         self._worker.check_connected()
 
         self._writer_registered = False
         self._reader_registered = False
 
+        if self._writer_channel is not None:
+            self._worker.core_worker.experimental_channel_register_reader_network(
```

Comment:
Does this work if there are multiple readers on the remote node?

Reply:
Done.